url47087446e9_kellrott_SparqlScape_tgz-pJ8-org_json_TestJ8/src/org/sparqlscape/LinkTriple.java

```diff
@@ -0,0 +1,43 @@
+package org.sparqlscape;
+
+public class LinkTriple implements Comparable<LinkTriple> {
+
+	public LinkTriple(String subj, String pred, String obj) {
+		this.subj = subj;
+		this.pred = pred;
+		this.obj = obj;
+	}
+	public String subj, pred, obj;
+
+	@Override
+	public String toString() {
+		return "<" + subj + "> <" + pred + "> <" + obj + ">";		
+	}	
+
+	@Override
+	public int compareTo(LinkTriple other) {
+		int diff1 = subj.compareTo(other.subj);		
+		int diff2 = pred.compareTo(other.pred);		
+		int diff3 = obj.compareTo(other.obj);				
+		if ( diff1 != 0 ) 
+			return diff1;
+		if ( diff2 != 0 ) 
+			return diff2;
+		if ( diff3 != 0 ) 
+			return diff3;
+		return 0;		
+	}
+
+	@Override
+	public int hashCode() {
+		return subj.hashCode() + pred.hashCode() + obj.hashCode();
+	}
+
+	public boolean equals(Object o) {
+		if (o instanceof LinkTriple) {
+			LinkTriple other = (LinkTriple) o;
+			return (subj == other.subj && pred == other.pred && obj == other.obj);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.